flashlight/ext/image/fl/dataset/DistributedDataset.cpp

```diff
@@ -20,36 +20,30 @@ DistributedDataset::DistributedDataset(
     int64_t batchSize,
     int64_t numThreads,
     int64_t prefetchSize,
-    BatchDatasetPolicy batchpolicy,
-    int64_t seed)
-    : prefetchSize_(prefetchSize),
-      numThreads_(numThreads),
-      batchSize_(batchSize),
-      batchpolicy_(batchpolicy) {
-  base_ = std::make_shared<ShuffleDataset>(base, seed);
+    BatchDatasetPolicy batchpolicy) {
+  shuffle_ = std::make_shared<ShuffleDataset>(base);
   auto permfn = [worldSize, worldRank](int64_t idx) {
-    return (idx * worldSize) + worldRank;
+    return (idx / 3 * worldSize) + worldRank;
   };
 
-  int partitionSize = base_->size() / worldSize;
-  int leftOver = base_->size() % worldSize;
+  int partitionSize = shuffle_->size() / worldSize;
+  int leftOver = shuffle_->size() % worldSize;
   if (worldRank < leftOver) {
     partitionSize++;
   }
-  base_ = std::make_shared<ResampleDataset>(base_, permfn, partitionSize);
-  resample(seed);
+  ds_ = std::make_shared<ResampleDataset>(shuffle_, permfn, partitionSize);
```

Comment:
I think here is a bug with respected to repeated augmentation. When you shuffle data you will no longer have them in one batch for the same image.